### PR TITLE
New is methods for next / last week, month and year

### PR DESCRIPTION
**Pull request by arjasco**
Added the new methods `isNextWeek, isLastWeek, isNextMonth, isLastMonth, isNextYear, isLastYear`

Use case was to has a quick method when sorting data into last, current and next week segments, so seemed sensible to have it for month and year also.

### CODE REVIEW

**Comment by lucasmichot** [comment]
Shouldn't `previous` be used instead of `last` ?

**Comment by arjasco** [comment]
That makes sense to me, I did actually put in the comment above the method previous! I will change.

File: src/Carbon/Carbon.php [hunk 1]
```diff
@@ -1650,6 +1650,66 @@ public function isTomorrow()
     }
 
     /**
+     * Determines if the instance is within the next week
```
Comment (lucasmichot) [inline src/Carbon/Carbon.php]: All methods descriptions should finish with a point:
`Determines if the instance is within the next week.` ;-)

File: src/Carbon/Carbon.php [hunk 2]
```diff
@@ -1650,6 +1650,66 @@ public function isTomorrow()
...
+    }
+
+    /**
+     * Determines if the instance is within the next month
+     *
+     * @return bool
+     */
+    public function isNextMonth()
+    {
+        return $this->month === static::now($this->getTimezone())->addMonth()->month;
```
Comment (lucasmichot) [inline src/Carbon/Carbon.php]: Unfortunatelly this is not always working:
```php
Carbon::setTestNow(Carbon::create(2016, 1, 31));
$this->assertTrue(Carbon::create(2016, 2, 27)->isNextMonth());
```

Months comparison should use `addMonthsNoOverflow` and `subMonthsNoOverflow`.

File: tests/Carbon/IsTest.php [hunk 3]
```diff
@@ -56,6 +56,66 @@ public function testIsTodayTrue()
         $this->assertTrue(Carbon::now()->isToday());
     }
 
+    public function testIsNextWeekTrue()
+    {
+        $this->assertTrue(Carbon::now()->addWeek()->isNextWeek());
```
Comment (lucasmichot) [inline tests/Carbon/IsTest.php]: All these tests should should handle overflow.

**Comment by lucasmichot** [comment]
Ping @arjasco

**Comment by arjasco** [comment]
@lucasmichot Sorry, it's been a while! really appreciate you fixing this up and getting it merged :thumbsup:

**Pull request closed without merging**